### ISSUE TITLE
github exporter: github_repo_release_downloads is absent most of the time

### ISSUE THREAD

**Issue by 532910** [grafana/agent#6494]
### What's wrong?

https://github.com/githubexporter/github-exporter/issues/52

no github_repo_release_downloads metric is provided

### Steps to reproduce

configure github exporter as described in documentation: https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.github/

### Software version

grafana-agent-flow 0.39.2-1

**Comment by tpaschalis** [comment]
Note: If this is indeed a bug with the upstream exporter (I haven't really looked at it yet), we'll have to fix it there first.

**Comment by hainenber** [comment]
It appears that GH's rate limiter got into action. I'm not sure on its aggression though. Maybe expired/invalid API token?

This is coming from my local machine trying to reproduce the issue.

The config is
```
prometheus.exporter.github "example" {
  repositories   = ["grafana/agent"]
  api_token      = "REDACTED"
}

// Configure a prometheus.scrape component to collect github metrics.
prometheus.scrape "demo" {
  targets    = prometheus.exporter.github.example.targets
  forward_to = [prometheus.remote_write.demo.receiver]
}

prometheus.remote_write "demo" {
  endpoint {
    url = "localhost:8428"
  }
}
```

<img width="1027" alt="image" src="https://github.com/grafana/agent/assets/41283691/6f24e0e3-9406-4808-bab5-1ad87cffef57">

**Comment by 532910** [comment]
> It appears that GH's rate limiter got into action. I'm not sure on its aggression though. Maybe expired/invalid API token?

This is not true: only one repo of several doesn't provide `github_repo_release_downloads`

```
prometheus.exporter.github "default" {
  api_token_file = "/etc/grafana-agent-github-token"
  organizations = [ "nspcc-dev" ]
  repositories = [
    "nspcc-dev/neo-go",
    "nspcc-dev/neofs-node",
    "nspcc-dev/neofs-rest-gw",
    "nspcc-dev/neofs-s3-gw",
    "nspcc-dev/neo-exporter",
   ]
}
```

`group by (repo) (github_repo_release_downloads)` gives:

```
{repo="neo-exporter"} 1
{repo="neo-go"} 1
{repo="neofs-rest-gw"} 1
{repo="neofs-s3-gw"} 1
```

**Comment by 532910** [comment]
needs-attention

**Comment by rfratto** [comment]
Hi there :wave: 

On April 9, 2024, Grafana Labs announced [Grafana Alloy](https://github.com/grafana/alloy), the spirital successor to Grafana Agent and the final form of Grafana Agent flow mode. As a result, Grafana Agent has been deprecated and will only be receiving bug and security fixes until its end-of-life around November 1, 2025. 

To make things easier for maintainers, we're in the process of migrating all issues tagged variant/flow to the Grafana Alloy repository to have a single home for tracking issues. This issue is likely something we'll want to address in both Grafana Alloy and Grafana Agent, so just because it's being moved doesn't mean we won't address the issue in Grafana Agent :)
